regex-assembly/exclude/unix-shell-fps-pl1.ra

```diff
@@ -0,0 +1,668 @@
+
+##! This list excludes command words that are prone to cause false
+##! positives from the following include files:
+##! - unix-shell-upto3.ra and
+##! - unix-shell-upto3-for-cmdline.ra
+##! - unix-shell-4andup-with-params.ra
+
+##! To automatically add exclusions for English words from the sources, run the following:
+##! original="$(grep -vE '^(?:#|$)' regex-assembly/exclude/unix-shell-fps-pl1.ra)"
+##! upto3="$(grep -vE '^(?:#|$)' regex-assembly/include/unix-shell-upto3.ra)"
+##! rest="$(grep -vE '^(?:#|$)' regex-assembly/include/unix-shell-4andup-with-params.ra)"
+##! english_upto3="$(util/fp-finder/spell.sh -m -e regex-assembly/include/unix-shell-upto3.ra)"
+##! # strip suffixes from words so that spell.sh works
+##! english_rest="$(util/fp-finder/spell.sh -m -e -s '{{[^}]+}}' regex-assembly/include/unix-shell-4andup-with-params.ra)"
+##! result=""
+##! function update_existing {
+##!   if [ -z "${1}" ]; then
+##!     return
+##!   fi
+##!   while read -r oword; do
+##!     found=0
+##!     while read -r eword; do
+##!       if grep -qE "^${eword}(?:[@~]|\{\{.*|$)" <<<"${oword}"; then
+##!         result="${result}${eword}\n"
+##!         result="${result}${eword}@\n"
+##!         result="${result}${eword}~\n"
+##!         result="${result}${eword}{{space-or-redirect}}\n"
+##!         result="${result}${eword}{{no-space}}\n"
+##!         found=1
+##!         break
+##!       fi
+##!     done <<<"${1}"
+##!     if [ ${found} -eq 0 ]; then
+##!       result="${result}${oword}\n"
+##!     fi
+##!   done <<<"${original}"
+##! }
+##! function add_new {
+##!   if [ -z "${1}" ]; then
+##!     return
+##!   fi
+##!   while read -r eword; do
+##!     if ! grep -qE "^${eword}(?:[@~]|\{\{.*|$)" <<<"${original}"; then
+##!       result="${result}${eword}\n"
+##!       result="${result}${eword}@\n"
+##!       result="${result}${eword}~\n"
+##!       result="${result}${eword}{{space-or-redirect}}\n"
+##!       result="${result}${eword}{{no-space}}\n"
+##!     fi
+##!   done <<<"${1}"
+##! }
+##! update_existing "${english_upto3}"
+##! update_existing "${english_rest}"
+##! add_new "${english_upto3}"
+##! add_new "${english_rest}"
+
+##! sed -i '' -e '61,$d' regex-assembly/exclude/unix-shell-fps-pl1.ra
+##! echo ${result} | sort | uniq >> regex-assembly/exclude/unix-shell-fps-pl1.ra
+
+ab
+ab@
+ab{{no-space}}
+ab{{space-or-redirect}}
+ab~
+adduser
+agetty
+alias{{space-or-redirect}}
+alpine{{space-or-redirect}}
+ansible-playbook
+apt
+apt-get
+apt@
+aptitude{{space-or-redirect}}
+apt{{no-space}}
+apt{{space-or-redirect}}
+apt~
+ar
+ar@
+arch{{space-or-redirect}}
+aria2c
+arj
+arp
+arp@
+arp{{no-space}}
+arp{{space-or-redirect}}
+arp~
+ar{{no-space}}
+ar{{space-or-redirect}}
+ar~
+as
+as@
+ascii-xfr
+ascii85
+ash
+ash@
+ash{{no-space}}
+ash{{space-or-redirect}}
+ash~
+aspell
+as{{no-space}}
+as{{space-or-redirect}}
+as~
+at
+at@
+atobm
+at{{no-space}}
+at{{space-or-redirect}}
+at~
+awk
+awk@
+awk{{space-or-redirect}}
```

Comment:
This file is used in multiple contexts to exclude words (lines). In one context the form might be `awk@`, in the other `awk{{space-or-redirect}}`. In that case it would suffice to list those two forms in this file but in some cases the forms will be, `awk~`, `awk{{no-space}}`. So I decided to simply list all possible forms for each word instead of having to synchronize the exact entries with this list. That makes it much simpler to generate.

Reply:
That makes sense. Thanks. Please add this information as comment.

Are you sure it's systematical? If no, I will check.

Reply:
Yes, I'm sure, the script does this:

```shell
# ...
         result="${result}${eword}\n"
         result="${result}${eword}@\n"
         result="${result}${eword}~\n"
         result="${result}${eword}{{space-or-redirect}}\n"
         result="${result}${eword}{{no-space}}\n"
# ...
```